internal/services/containerapps/container_app_environment_resource.go

```diff
@@ -478,6 +496,14 @@ func (r ContainerAppEnvironmentResource) CustomizeDiff() sdk.ResourceFunc {
 				}
 			}
 
+			if env.LogsDestination == "log-analytics" && env.LogAnalyticsWorkspaceId == "" {
+				return fmt.Errorf("`log_analytics_workspace_id` must be set when `logs_destination` is set to `log-analytics`")
+			}
+
+			if env.LogAnalyticsWorkspaceId != "" && env.LogsDestination != "log-analytics" {
+				return fmt.Errorf("`log_analytics_workspace_id` can only be set when `logs_destination` is set to `log-analytics`")
+			}
```

Comment:
(related to above) if we infer the value for this field based on `log_analytics_workspace_id` having a value, then we could instead remove this logic?